Create tt packages for 64bit ARM (aarch64) Linux

If cross compilation will be used, then you need to replace Dup2 with Dup3
otherwise there will be an error:

```sh
$ GOARCH=arm64 mage build
Building release tt...
# github.com/tarantool/tt/cli/build
cli/build/local.go:88:20: undefined: syscall.Dup2
cli/build/local.go:91:17: undefined: syscall.Dup2
cli/build/local.go:101:20: undefined: syscall.Dup2
```

```diff
--- a/cli/build/local.go
+++ b/cli/build/local.go
@@ -85,10 +85,10 @@ func buildLocal(cmdCtx *cmdcontext.CmdCtx, cliOpts *config.CliOpts, buildCtx *Bu
}
defer devNull.Close()

- if err = syscall.Dup2(int(devNull.Fd()), syscall.Stdout); err != nil {
+ if err = syscall.Dup3(int(devNull.Fd()), syscall.Stdout, 0); err != nil {
return err
}
- defer syscall.Dup2(savedStdoutFd, syscall.Stdout)
+ defer syscall.Dup3(savedStdoutFd, syscall.Stdout, 0)
}

rocksMakeCmd := []string{"make"}
@@ -98,7 +98,7 @@ func buildLocal(cmdCtx *cmdcontext.CmdCtx, cliOpts *config.CliOpts, buildCtx *Bu
if err := rocks.Exec(cmdCtx, cliOpts, rocksMakeCmd); err != nil {
return err
}
- if err := syscall.Dup2(savedStdoutFd, syscall.Stdout); err != nil {
+ if err := syscall.Dup3(savedStdoutFd, syscall.Stdout, 0); err != nil {
return err
```

--- a/.github/actions/setup-etcd/action.yml
+++ b/.github/actions/setup-etcd/action.yml
@@ -21,7 +21,8 @@ runs:
         rm -rf ${INSTALL_PREFIX} && mkdir -p ${INSTALL_PREFIX}
 
         OS_NAME="$(uname | tr '[:upper:]' '[:lower:]')"
-        FILENAME="etcd-${ETCD_VER}-${OS_NAME}-amd64"
+        ARCH=$(uname -m | awk '{print ($0 == "x86_64")?"amd64":"arm64"}')
+        FILENAME="etcd-${ETCD_VER}-${OS_NAME}-${ARCH}"
         if [ "${OS_NAME}" == "linux" ]; then
           curl -L "${BASE_URL}/${ETCD_VER}/${FILENAME}.tar.gz" -o "${INSTALL_PREFIX}/${FILENAME}.tar.gz"
           tar xvzf "${INSTALL_PREFIX}/${FILENAME}.tar.gz" -C ${INSTALL_PREFIX} --strip-components=1

--- a/.github/workflows/full-ci.yml
+++ b/.github/workflows/full-ci.yml
@@ -52,6 +52,45 @@ jobs:
       - name: Integration tests
         run: mage integrationfull
 
+  full-ci-ce-linux-arm64:
+    if: |
+      (github.event_name == 'push') ||
+      (github.event_name == 'pull_request' &&
+        github.event.action == 'labeled' &&
+        github.event.label.name == 'full-ci') ||
+      (github.event_name == 'pull_request' &&
+        github.event.action == 'synchronize' &&
+        contains(github.event.pull_request.labels.*.name, 'full-ci'))
+    runs-on: graviton
+    strategy:
+      fail-fast: false
+    container:
+      image: tarantool/testing:tt-build
+      options: '--init --privileged'
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Install test requirements
+        run: |
+          pip3 install setuptools==59.5.0
+          pip3 install -r test/requirements.txt
+        shell: bash
+
+      - name: Install etcd
+        uses: ./.github/actions/setup-etcd
+
+      - name: Build test
+        run: mage build
+
+      - name: Unit tests
+        run: mage unit:fullSkipDocker
+
+      - name: Integration tests
+        run: mage integrationfullskipdocker
+
   full-ci-sdk:
     # Tests will run only when the pull request is labeled with `full-ci`. To
     # avoid security problems, the label must be reset manually for every run.

--- a/.github/workflows/publish.yml
+++ b/.github/workflows/publish.yml
@@ -73,6 +73,52 @@ jobs:
           name: packages-linux
           path: dist
 
+  create-packages-linux-arm64:
+    runs-on: graviton
+    strategy:
+      fail-fast: false
+    container:
+      image: tarantool/testing:tt-build
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Install etcd
+        uses: ./.github/actions/setup-etcd
+
+      - name: Set GoReleaser flags
+        id: set-goreleaser-flags
+        run: |
+          if ${{ startsWith(github.ref, 'refs/tags') }} ; then
+            echo "::set-output name=GORELEASER_FLAGS::--rm-dist --skip-validate"
+          else
+            echo "::set-output name=GORELEASER_FLAGS::--rm-dist --snapshot"
+          fi
+
+      # These scripts will be picked up while building packages with goreleaser.
+      - name: Generate ZSH and Bash completion scripts
+        run: |
+          mage build
+          ./tt completion bash > tt-completion.bash
+          ./tt completion zsh > tt-completion.zsh
+
+      - name: Build packages
+        env:
+          CGO_LDFLAGS: "-L/opt/openssl/lib"
+          CGO_CFLAGS: "-I/opt/openssl/include"
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          goreleaser release ${{ steps.set-goreleaser-flags.outputs.GORELEASER_FLAGS }} \
+            --skip-publish --config ./goreleaser/.goreleaser_linux_arm64.yml
+
+      - name: Upload packages artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: packages-linux-arm64
+          path: dist
+
   create-packages-macos:
     runs-on: macos-latest
     steps:
@@ -128,7 +174,7 @@ jobs:
           path: dist
 
   draft-release:
-    needs: [create-packages-linux, create-packages-macos]
+    needs: [create-packages-linux, create-packages-linux-arm64, create-packages-macos]
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@master
@@ -143,6 +189,7 @@ jobs:
       - name: Make directories
         run: |
           mkdir -p ./dist-prebuilt/packages-linux
+          mkdir -p ./dist-prebuilt/packages-linux-arm64
           mkdir -p ./dist-prebuilt/packages-macos
 
       - name: Download linux binaries
@@ -150,6 +197,12 @@ jobs:
         with:
           name: packages-linux
           path: ./dist-prebuilt/packages-linux
+
+      - name: Download linux arm64 binaries
+        uses: actions/download-artifact@v3
+        with:
+          name: packages-linux-arm64
+          path: ./dist-prebuilt/packages-linux-arm64
 
       - name: Download macos binaries
         uses: actions/download-artifact@v3
@@ -162,11 +215,15 @@ jobs:
           rm ./dist-prebuilt/packages-linux/artifacts.json
           rm ./dist-prebuilt/packages-linux/config.yaml
           rm ./dist-prebuilt/packages-linux/metadata.json
+          rm ./dist-prebuilt/packages-linux-arm64/artifacts.json
+          rm ./dist-prebuilt/packages-linux-arm64/config.yaml
+          rm ./dist-prebuilt/packages-linux-arm64/metadata.json
           rm ./dist-prebuilt/packages-macos/artifacts.json
           rm ./dist-prebuilt/packages-macos/config.yaml
           rm ./dist-prebuilt/packages-macos/metadata.json
           rm ./dist-prebuilt/packages-macos/tt_darwin_amd64_v1/tt
           rm ./dist-prebuilt/packages-linux/tt_linux_amd64_v1/tt
+          rm ./dist-prebuilt/packages-linux-arm64/tt_linux_arm64/tt
 
       - name: Pack additional archives
         run: |
@@ -177,8 +234,10 @@ jobs:
         run: |
           cd ./dist-prebuilt
           cat ./packages-linux/tt*checksums.txt >> checksums.txt
+          cat ./packages-linux-arm64/tt*checksums.txt >> checksums.txt
           cat ./packages-macos/tt*checksums.txt >> checksums.txt
           rm ./packages-linux/tt*checksums.txt
+          rm ./packages-linux-arm64/tt*checksums.txt
           rm ./packages-macos/tt*checksums.txt
           mv checksums.txt ./packages-macos/
 
@@ -199,7 +258,7 @@ jobs:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
 
   publish-s3:
-    needs: create-packages-linux
+    needs: [create-packages-linux, create-packages-linux-arm64]
     runs-on: ubuntu-latest
     if: startsWith(github.ref, 'refs/tags')
     steps:
@@ -216,13 +275,25 @@ jobs:
           cd mage
           go run bootstrap.go
 
-      - name: Download packages artifacts
+      - name: Download x86_64 packages artifacts
         uses: actions/download-artifact@v2
         with:
           name: packages-linux
           path: dist
 
-      - name: Publish packages to RWS
+      - name: Publish x86_64 packages to RWS
+        env:
+          RWS_URL_PART: https://rws.tarantool.org/release/modules
+          RWS_AUTH: ${{ secrets.RWS_AUTH }}
+        run: mage publishRWS && rm -rf dist
+
+      - name: Download arm64 packages artifacts
+        uses: actions/download-artifact@v2
+        with:
+          name: packages-linux-arm64
+          path: dist
+
+      - name: Publish arm64 packages to RWS
         env:
           RWS_URL_PART: https://rws.tarantool.org/release/modules
           RWS_AUTH: ${{ secrets.RWS_AUTH }}

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -49,6 +49,42 @@ jobs:
       - name: Integration tests
         run: mage integration
 
+  tests-ce-linux-arm64:
+    if: |
+      (github.event_name == 'push') ||
+      (github.event_name == 'pull_request' &&
+        github.event.pull_request.head.repo.owner.login != 'tarantool' &&
+        !contains(github.event.pull_request.labels.*.name, 'full-ci'))
+    runs-on: graviton
+    strategy:
+      fail-fast: false
+    container:
+      image: tarantool/testing:tt-build
+      options: '--init --privileged'
+    steps:
+      - uses: actions/checkout@master
+        with:
+          fetch-depth: 0
+          submodules: recursive
+
+      - name: Install test requirements
+        run: |
+          pip3 install setuptools==59.5.0
+          pip3 install -r test/requirements.txt
+        shell: bash
+
+      - name: Install etcd
+        uses: ./.github/actions/setup-etcd
+
+      - name: Build test
+        run: mage build
+
+      - name: Unit tests
+        run: mage unit
+
+      - name: Integration tests
+        run: mage integration
+
   tests-ee:
     # The same as for tests-ce, but it does not run on pull requests from
     # forks by default. Tests will run only when the pull request is labeled

--- a/ci/Dockerfile
+++ b/ci/Dockerfile
@@ -1,0 +1,80 @@
+# Docker file used in workflows:
+# full-ci-ce-linux-arm64, create-packages-linux-arm64, tests-ce-linux-arm64
+FROM ubuntu:20.04
+
+ARG DEBIAN_FRONTEND=noninteractive
+ENV TZ=Etc/UTC
+
+RUN apt update \
+    && apt -y upgrade \
+    && apt install --no-install-suggests -y \
+        vim \
+        git \
+        cpio \
+        wget \
+        curl \
+        rsync \
+        golang \
+        make \
+        sudo \
+        gcc \
+        cmake \
+        unzip \
+        zip \
+        build-essential \
+        ninja-build \
+        lua5.1 \
+        luarocks \
+        lcov \
+        ruby-dev \
+        liblz4-dev \
+        autoconf \
+        automake \
+        libtool \
+        libssl-dev \
+        python3-pytest \
+        python3-psutil \
+        python3-pip \
+        lsb-release \
+        software-properties-common \
+    && apt clean
+
+RUN add-apt-repository ppa:longsleep/golang-backports \
+    && apt update \
+    && apt install -y golang \
+    && apt clean
+
+RUN if [ "$(uname -m)" = "x86_64" ]; \
+        then ARCHITECTURE="64bit"; \
+    elif [ "$(uname -m)" = "aarch64" ]; \
+        then ARCHITECTURE="ARM64"; \
+    else \
+        echo "Unsupported platform!"; exit 0; \
+    fi \
+    && wget -P /tmp/ "https://github.com/magefile/mage/releases/download/v1.15.0/mage_1.15.0_Linux-${ARCHITECTURE}.tar.gz" \
+    && tar xzf "/tmp/mage_1.15.0_Linux-${ARCHITECTURE}.tar.gz" -C /usr/local/bin/ \
+    && rm -rf /tmp/*
+
+RUN echo 'deb [trusted=yes] https://repo.goreleaser.com/apt/ /' > /etc/apt/sources.list.d/goreleaser.list \
+    && apt update \
+    && apt install -y goreleaser=1.12.3 \
+    && apt clean
+
+RUN wget -P /tmp/ https://github.com/openssl/openssl/releases/download/openssl-3.0.8/openssl-3.0.8.tar.gz \
+    && tar -xvf /tmp/openssl-3.0.8.tar.gz -C /tmp/ \
+    && cd /tmp/openssl-3.0.8 \
+    && ./Configure --prefix=/opt/openssl no-shared \
+    && make -j \ 
+    && make install \
+    && rm -rf /tmp/*
+
+RUN ETCD_VER=v3.5.9 \
+    FILENAME="etcd-${ETCD_VER}-linux-$(uname -m | awk '{print ($0 == "x86_64")?"amd64":"arm64"}')" \
+    && mkdir -p /opt/etcd/ \
+    && curl -L "https://github.com/etcd-io/etcd/releases/download/${ETCD_VER}/${FILENAME}.tar.gz" -o "/opt/etcd/${FILENAME}.tar.gz" \
+    && tar xvzf "/opt/etcd/${FILENAME}.tar.gz" -C /opt/etcd --strip-components=1 \
+    && rm /opt/etcd/*.tar.gz
+
+RUN curl -L https://tarantool.io/iqJapRm/release/3/installer.sh | bash \
+    && apt install -y tarantool tarantool-dev \
+    && apt clean    

--- a/cli/create/internal/steps/move_app_dir_test.go
+++ b/cli/create/internal/steps/move_app_dir_test.go
@@ -3,6 +3,7 @@ package steps
 import (
 	"fmt"
 	"os"
+	"os/user"
 	"path/filepath"
 	"testing"
 
@@ -58,6 +59,9 @@ func TestMoveAppDirSourceDirMissing(t *testing.T) {
 }
 
 func TestMoveAppDirTargetDirRemovalFailure(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
 	var createCtx create_ctx.CreateCtx
 	templateCtx := app_template.NewTemplateContext()
 

--- a/cli/docker/docker_integration_test.go
+++ b/cli/docker/docker_integration_test.go
@@ -1,4 +1,4 @@
-//go:build integration
+//go:build integration_docker
 
 package docker
 

--- a/cli/install/install_test.go
+++ b/cli/install/install_test.go
@@ -2,6 +2,7 @@ package install
 
 import (
 	"os"
+	"os/user"
 	"path/filepath"
 	"testing"
 
@@ -11,6 +12,9 @@ import (
 )
 
 func Test_dirsAreWriteable(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
 	tmpDirNonWriteableForAll := t.TempDir()
 	// dr-xr-xr-x mode.
 	permissions := 0555
@@ -49,6 +53,9 @@ func Test_dirsAreWriteable(t *testing.T) {
 }
 
 func Test_subDirIsWritable(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
 	tmpDirNonWriteableForAll := t.TempDir()
 	// dr-xr-xr-x mode.
 	permissions := 0555

--- a/cli/running/running_test.go
+++ b/cli/running/running_test.go
@@ -3,6 +3,7 @@ package running
 import (
 	"io"
 	"os"
+	"os/user"
 	"path/filepath"
 	"testing"
 
@@ -25,6 +26,9 @@ func (mock *mockRepository) ValidateAll() error {
 }
 
 func Test_CollectInstances(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
 	instancesEnabledPath := filepath.Join("testdata", "instances_enabled")
 
 	instances, err := CollectInstances("script", instancesEnabledPath,

--- a/cli/util/util_test.go
+++ b/cli/util/util_test.go
@@ -5,6 +5,7 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"os/user"
 	"path/filepath"
 	"regexp"
 	"strings"
@@ -85,6 +86,9 @@ func TestIsRegularFile(t *testing.T) {
 }
 
 func TestCreateDirectory(t *testing.T) {
+	if user, err := user.Current(); err == nil && user.Uid == "0" {
+		t.Skip("Skipping the test, it shouldn't run as root")
+	}
 	tempDir := t.TempDir()
 	require.NoError(t, os.Mkdir(filepath.Join(tempDir, "dir1"), 0750))
 

--- a/goreleaser/.goreleaser_linux_arm64.yml
+++ b/goreleaser/.goreleaser_linux_arm64.yml
@@ -1,0 +1,150 @@
+before:
+  hooks:
+    - go mod download
+
+env:
+  - GO111MODULE=on
+
+builds:
+  -
+    id: tt
+    binary: tt
+    dir: cli
+
+    env:
+      - CGO_ENABLED=1
+
+    tags:
+      - netgo
+      - osusergo
+      - openssl_static
+
+    ldflags:
+      - -linkmode=external -extldflags -static
+      - -s -w
+      - -X github.com/tarantool/tt/cli/version.gitTag={{ .Tag }}
+      - -X github.com/tarantool/tt/cli/version.gitCommit={{ .ShortCommit }}
+      - -X github.com/tarantool/tt/cli/configure.defaultConfigPath=/etc/tarantool
+
+    goos:
+      - linux
+    goarch:
+      - arm64
+
+    hooks:
+      pre:
+        - mage PatchCC
+        - mage GenerateGoCode
+
+archives:
+  -
+    id: "tt"
+    builds: ['tt']
+    format: tar.gz
+    name_template: "{{ .ProjectName }}-{{ .Version }}-{{ .Os }}-{{ .Arch }}"
+    files:
+      - README.md
+      - LICENSE
+      - CHANGELOG.md
+
+snapshot:
+  name_template: "{{ .Tag }}-{{ .ShortCommit }}"
+
+nfpms:
+  -
+    id: "tt"
+    builds: ['tt']
+    formats:
+      - deb
+      - rpm
+    homepage: "https://github.com/tarantool/tt"
+    maintainer: "Leonid Vasiliev <lvasiliev@tarantool.org>"
+    description: "Command-line utility to manage Tarantool applications"
+    license: "BSD-2 MIT Apache-2.0 BSD MPL-2.0"
+    recommends:
+      - git
+      - zip
+      - unzip
+      - curl
+    contents:
+      - src: "package/tt.yaml.default"
+        dst: "/etc/tarantool/tt.yaml"
+        type: config
+
+      - src: "package/logrotate.conf"
+        dst: "/etc/logrotate.d/tt.conf"
+        type: config
+        file_info:
+          mode: 0600
+
+      - src: package/instances.available
+        dst: /etc/tarantool/instances.available
+        type: tree
+      - dst: /etc/tarantool/instances.enabled
+        type: dir
+        file_info:
+          mode: 0755
+      - dst: /etc/tarantool
+        type: ghost
+
+      # Artifacts directories for the system tt environment.
+      - dst: /var/lib/tarantool/sys_env
+        type: dir
+        file_info:
+          mode: 0750
+          owner: tarantool
+          group: tarantool
+      - dst: /var/lib/tarantool
+        type: ghost
+
+      - dst: /var/run/tarantool/sys_env
+        type: dir
+        file_info:
+          mode: 0750
+          owner: tarantool
+          group: tarantool
+      - dst: /var/run/tarantool
+        type: ghost
+
+      - dst: /var/log/tarantool/sys_env
+        type: dir
+        file_info:
+          mode: 0750
+          owner: tarantool
+          group: tarantool
+      - dst: /var/log/tarantool
+        type: ghost
+
+      # According to systemd.unit manual page, location for system mode units installed by package
+      # manager is different for deb-based and rpm-based distributions.
+      - src: package/systemd/replicaset_example@.service
+        dst: /lib/systemd/system/replicaset_example@.service
+        packager: deb
+      - src: package/systemd/replicaset_example@.service
+        dst: /usr/lib/systemd/system/replicaset_example@.service
+        packager: rpm
+
+      # ZSH and Bash autocompletion scripts get placed in appropriate directories
+      # and get activated automatically with restarting corresponding shell after
+      # the package installation.
+      # Sources:
+      # https://github.com/scop/bash-completion/blob/master/README.md#faq
+      # https://github.com/zsh-users/zsh-completions/blob/master/zsh-completions-howto.org#intro
+      - src: "tt-completion.bash"
+        dst: "/usr/share/bash-completion/completions/tt"
+      - src: "tt-completion.zsh"
+        dst: "/usr/share/zsh/vendor-completions/_tt"
+
+    overrides:
+      rpm:
+        file_name_template: >-
+          {{- .ProjectName }}-
+          {{- .Version }}.
+          {{- if eq .Arch "arm64" }}aarch64
+          {{- else }}{{ .Arch }}{{ end }}
+
+      deb:
+        file_name_template: "{{ .ProjectName }}_{{ .Version }}_{{ .Arch }}"
+
+    scripts:
+      preinstall: "package/scripts/preinstall.sh"

--- a/magefile.go
+++ b/magefile.go
@@ -347,6 +347,13 @@ func (Unit) Default() error {
 func (Unit) Full() error {
 	fmt.Println("Running full unit tests...")
 
+	return runUnitTests([]string{"-tags", "integration,integration_docker"})
+}
+
+// Run unit tests with a Tarantool instance integration, excluding docker tests.
+func (Unit) FullSkipDocker() error {
+	fmt.Println("Running full unit tests, excluding docker...")
+
 	return runUnitTests([]string{"-tags", "integration"})
 }
 
@@ -375,7 +382,7 @@ func (Unit) Coverage() error {
 	}
 
 	err = runUnitTests([]string{
-		"-tags", "integration",
+		"-tags", "integration,integration_docker",
 		"-cover",
 		"-args", fmt.Sprintf(`-test.gocoverdir=%s`, coverDir)})
 	if err != nil {
@@ -407,6 +414,14 @@ func IntegrationFull() error {
 
 	return sh.RunV(pythonExecutableName, "-m", "pytest", "-m", "not slow_ee and not notarantool",
 		"test/integration")
+}
+
+// Run full set of integration tests, excluding docker tests.
+func IntegrationFullSkipDocker() error {
+	fmt.Println("Running all integration tests, excluding docker...")
+
+	return sh.RunV(pythonExecutableName, "-m", "pytest", "-m",
+		"not slow_ee and not notarantool and not docker", "test/integration")
 }
 
 // Run set of ee integration tests.

--- a/magefile.publish.go
+++ b/magefile.publish.go
@@ -71,7 +71,7 @@ func walkMatch(root string, patterns []string) ([]string, error) {
 func getPatterns(distro Distro) ([]string, error) {
 
 	if distro.OS == "el" || distro.OS == "fedora" {
-		return []string{"*.x86_64.rpm", "*.noarch.rpm"}, nil
+		return []string{"*.rpm", "*.noarch.rpm"}, nil
 	}
 
 	if distro.OS == "ubuntu" || distro.OS == "debian" {

--- a/test/integration/init/test_init.py
+++ b/test/integration/init/test_init.py
@@ -2,6 +2,7 @@ import os
 import shutil
 import subprocess
 
+import pytest
 import yaml
 
 from utils import config_name
@@ -305,6 +306,10 @@ def test_tarantoolctl_cfg_from_doc(tt_cmd, tmpdir):
     check_env_dirs(tmpdir, "instances")
 
 
+@pytest.mark.skipif(
+    os.getuid() == 0,
+    reason="Skipping the test, it shouldn't run as root",
+)
 def test_init_tarantoolctl_app_no_read_permissions(tt_cmd, tmpdir):
     shutil.copy(os.path.join(os.path.dirname(__file__), "configs", "tarantoolctl.lua"),
                 os.path.join(tmpdir, ".tarantoolctl"))

--- a/test/integration/install/test_install.py
+++ b/test/integration/install/test_install.py
@@ -94,7 +94,7 @@ def test_install_uninstall_tt_specific_commit(tt_cmd, tmpdir):
         f.write('env:\n  bin_dir:\n  inc_dir:\n')
 
     # Install specific tt's commit.
-    install_cmd = [tt_cmd, "--cfg", configPath, "install", "tt", "400167c"]
+    install_cmd = [tt_cmd, "--cfg", configPath, "install", "tt", "97c7b73"]
     instance_process = subprocess.Popen(
         install_cmd,
         cwd=tmpdir,
@@ -116,11 +116,11 @@ def test_install_uninstall_tt_specific_commit(tt_cmd, tmpdir):
         text=True
     )
     start_output = installed_program_process.stdout.readline()
-    assert re.search(r"Tarantool CLI version 1.3.0", start_output)
-    assert re.search(r"commit: 400167c", start_output)
+    assert re.search(r"Tarantool CLI version 2.1.1", start_output)
+    assert re.search(r"commit: 97c7b73", start_output)
 
     # Uninstall specific tt's commit.
-    uninstall_cmd = [tt_cmd, "--cfg", configPath, "uninstall", "tt", "400167c"]
+    uninstall_cmd = [tt_cmd, "--cfg", configPath, "uninstall", "tt", "97c7b73"]
     uninstall_instance_process = subprocess.Popen(
         uninstall_cmd,
         cwd=tmpdir,
@@ -131,8 +131,8 @@ def test_install_uninstall_tt_specific_commit(tt_cmd, tmpdir):
     first_output = uninstall_instance_process.stdout.readline()
     assert re.search(r"Removing binary...", first_output)
     second_output = uninstall_instance_process.stdout.readline()
-    assert re.search(r"tt=400167c is uninstalled", second_output)
-    assert not os.path.exists(os.path.join(tmpdir, "bin", "tt_400167c"))
+    assert re.search(r"tt=97c7b73 is uninstalled", second_output)
+    assert not os.path.exists(os.path.join(tmpdir, "bin", "tt_97c7b73"))
 
 
 @pytest.mark.slow
@@ -187,7 +187,7 @@ def test_install_tt_specific_version(tt_cmd, tmpdir):
         f.write('env:\n  bin_dir:\n  inc_dir:\n')
 
     # Install latest tt.
-    install_cmd = [tt_cmd, "--cfg", configPath, "install", "tt", "1.0.0"]
+    install_cmd = [tt_cmd, "--cfg", configPath, "install", "tt", "2.1.2"]
     instance_process = subprocess.Popen(
         install_cmd,
         cwd=tmpdir,
@@ -210,7 +210,7 @@ def test_install_tt_specific_version(tt_cmd, tmpdir):
         text=True
     )
     start_output = installed_program_process.stdout.readline()
-    assert re.search(r"Tarantool CLI version 1.0.0", start_output)
+    assert re.search(r"Tarantool CLI version 2.1.2", start_output)
 
 
 @pytest.mark.slow
@@ -301,6 +301,7 @@ def test_install_tarantool(tt_cmd, tmpdir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_install_tarantool_in_docker(tt_cmd, tmpdir):
     if platform.system() == "Darwin":
         pytest.skip("/set platform is unsupported")

--- a/test/integration/pack/test_pack.py
+++ b/test/integration/pack/test_pack.py
@@ -1015,6 +1015,7 @@ def verify_rpmdeb_package_content(pkg_dir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_pack_deb(tt_cmd, tmpdir):
     if shutil.which('docker') is None:
         pytest.skip("docker is not installed in this system")
@@ -1092,6 +1093,7 @@ def test_pack_deb(tt_cmd, tmpdir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_pack_rpm(tt_cmd, tmpdir):
     if shutil.which('docker') is None:
         pytest.skip("docker is not installed in this system")
@@ -1212,6 +1214,7 @@ def test_pack_rpm_use_docker(tt_cmd, tmpdir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_pack_deb_use_docker_tnt_version(tt_cmd, tmpdir):
     if shutil.which('docker') is None:
         pytest.skip("docker is not installed in this system")
@@ -1251,6 +1254,7 @@ def test_pack_deb_use_docker_tnt_version(tt_cmd, tmpdir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_pack_rpm_use_docker_wrong_version_format(tt_cmd, tmpdir):
     if shutil.which('docker') is None:
         pytest.skip("docker is not installed in this system")
@@ -1276,6 +1280,7 @@ def test_pack_rpm_use_docker_wrong_version_format(tt_cmd, tmpdir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_pack_rpm_use_docker_wrong_version(tt_cmd, tmpdir):
     if shutil.which('docker') is None:
         pytest.skip("docker is not installed in this system")
@@ -1348,6 +1353,7 @@ def test_pack_deb_use_docker(tt_cmd, tmpdir):
 
 
 @pytest.mark.slow
+@pytest.mark.docker
 def test_pack_rpm_with_pre_and_post_inst(tt_cmd, tmpdir):
     if shutil.which('docker') is None:
         pytest.skip("docker is not installed in this system")

--- a/test/pytest.ini
+++ b/test/pytest.ini
@@ -5,6 +5,7 @@ markers =
     slow: marks tests as slow (deselect with '-m "not slow"')
     slow_ee: marks ee tests as slow (deselect with '-m "not slow_ee"')
     notarantool: marks tests that must be invoked without pre-installed tarantool (deselect with '-m "not notarantool"')
+    docker: marks tests that use docker (deselect with '-m "not docker"')
 
 # Max test duration is 15 minutes.
 timeout = 900

